src/porting_android.cpp

```diff
@@ -291,19 +270,17 @@ v2u32 getDisplaySize()
 		jmethodID getDisplayWidth = jnienv->GetMethodID(nativeActivity,
 				"getDisplayWidth", "()I");
 
-		if (getDisplayWidth == 0) {
-			assert("porting::getDisplayWidth unable to find java getDisplayWidth method" == 0);
-		}
+		if (getDisplayWidth == nullptr)
+			assert("porting::getDisplayWidth unable to find java getDisplayWidth method" == nullptr);
 
 		retval.X = jnienv->CallIntMethod(app_global->activity->clazz,
 				getDisplayWidth);
 
 		jmethodID getDisplayHeight = jnienv->GetMethodID(nativeActivity,
 				"getDisplayHeight", "()I");
 
-		if (getDisplayHeight == 0) {
-			assert("porting::getDisplayHeight unable to find java getDisplayHeight method" == 0);
-		}
+		if (getDisplayHeight == nullptr)
+			assert("porting::getDisplayHeight unable to find java getDisplayHeight method" == nullptr);
```

Comment:
wtf is that ? comparing a string and nullptr ?

Reply:
`getDisplayHeight` is int, not string.
if we do not get the values ​​from Java, it will be NULL

Reply:
yep but assert require a boolean, and here it's always false... which will assert yes, but not a very nice way. Can you use FATAL_ERROR_IF macro instead to have both the message and the condition ?

Reply:
```c++
assert(getDisplayHeight && "porting::getDisplayHeight unable to find java getDisplayHeight method");
```

is the standard C++ way to do that. Using our own macros would be better

> wtf is that ? comparing a string and nullptr ?

A `const char*` can be compared to nullptr, however it'll never be nullptr in this case.

Reply:
I would generally delete this. I have never seen a crash with a this error. Is it if someone intentionally removes a method from java activity